docs/pages/guides/config-plugins.md

```diff
@@ -0,0 +1,614 @@
+
+Plugins will generally have `@expo/config-plugins` installed as a dependency, and `expo-module-scripts`, `@expo/config-types` installed as a devDependencies.
+
+### Setting up a playground environment
+
+You can develop plugins easily using JS, but if you want to setup Jest tests and use TypeScript, you're gonna want a monorepo.
+
+A monorepo will enable you to work on a node module and import it in your Expo app like you would if it were published to NPM. Expo config plugins have full monorepo support built-in so all you need to do is setup a project.
+
+We recommend using [`expo-yarn-workspaces`](https://www.npmjs.com/package/expo-yarn-workspaces) which makes Expo monorepos very easy to setup.
+In your monorepo's `packages/` folder, create a module, and [bootstrap a config plugin](https://github.com/expo/expo/tree/master/packages/expo-module-scripts#-config-plugin) in it.
+
+### Modifying the AndroidManifest.xml
+
+You can use built-in types and helpers to ease the process of working with complex objects.
+Here's an example of adding a `<meta-data android:name="..." android:value="..."/>` to the default `<application android:name=".MainApplication" />`.
+
+```ts
+import { AndroidConfig, ConfigPlugin, withAndroidManifest } from '@expo/config-plugins';
+import { ExpoConfig } from '@expo/config-types';
+
+// Using helpers keeps error messages unified and helps cut down on XML format changes.
+const { addMetaDataItemToMainApplication, getMainApplicationOrThrow } = AndroidConfig.Manifest;
+
+export const withMyCustomConfig: ConfigPlugin = config => {
+  return withAndroidManifest(config, async config => {
+    // Modifiers can be async, but try to keep them fast.
+    config.modResults = await setCustomConfigAsync(config, config.modResults);
+    return config;
+  });
+};
+
+// Splitting this function out of the mod makes it easier to test.
+async function setCustomConfigAsync(
+  config: Pick<ExpoConfig, 'android'>,
+  androidManifest: AndroidConfig.Manifest.AndroidManifest
+): Promise<AndroidConfig.Manifest.AndroidManifest> {
+  const appId = 'my-app-id';
+  // Get the <application /> tag and assert if it doesn't exist.
+  const mainApplication = getMainApplicationOrThrow(androidManifest);
+
+  addMetaDataItemToMainApplication(
+    mainApplication,
+    // value for `android:name`
+    'my-app-id-key',
+    // value for `android:value`
+    appId
+  );
+
+  return androidManifest;
+}
+```
+
+### Modifying the Info.plist
+
+Using the `withInfoPlist` is a bit safer than statically modifying the `expo.ios.infoPlist` object in the app.json because it reads the contents of the Info.plist and merges it with the `expo.ios.infoPlist`, this means you can attempt to keep your changes from being overwritten.
```

Comment:
```suggestion
Using the `withInfoPlist` is a bit safer than statically modifying the `expo.ios.infoPlist` object in the `app.json` because it reads the contents of the Info.plist and merges it with the `expo.ios.infoPlist`, this means you can attempt to keep your changes from being overwritten.
```